### DISCUSSION TITLE
Bidirection ManyToMany that only works one way

### DISCUSSION THREAD

**Discussion by toto33331** [doctrine/doctrine2#9573] [Support Questions]
Hi,
I hope I post on the good place. **I have a bidirectional ManyToMany that only works on one side** (the owning side ? IDK)
Here is some code to explain:

class GlobalContent :
```
    /**
    * @ORM\ManyToMany(targetEntity="App\Entity\Groupement", inversedBy="all_global_contents")
    * @ORM\JoinTable(name="global_content_groupement")
    */
    private $all_groupements;

    /**
     * @return Collection|Groupement[]
     */
    public function getAllGroupements(): Collection
    {
        return $this->all_groupements;
    }

    public function addAllGroupement(Groupement $groupement): self
    {
        if (!$this->all_groupements->contains($groupement)) {
            $this->all_groupements[] = $groupement;
        }

        return $this;
    }

    public function removeAllGroupement(Groupement $groupement): self
    {
        if ($this->all_groupements->contains($groupement)) {
            $this->all_groupements->removeElement($groupement);
        }

        return $this;
    }
```

class Groupement:
```
/**
* @ORM\ManyToMany(targetEntity="App\Entity\GlobalContent", mappedBy="all_groupements")
*/
private $all_global_contents;

    /**
     * @return Collection|GlobalContent[]
     */
    public function getAllGlobalContents(): Collection
    {
        return $this->all_global_contents;
    }

    public function addAllGlobalContent(GlobalContent $global_content): self
    {
        if (!$this->all_global_contents->contains($global_content)) {
            $this->all_global_contents[] = $global_content;
        }
        return $this;
    }

    public function removeAllGlobalContent(GlobalContent $global_content): self
    {
        if ($this->all_global_contents->contains($global_content)) {
            $this->all_global_contents->removeElement($global_content);
        }
        return $this;
    }
```

This code as it is only works if I add/remove some Groupment from a GlobalContent object. From a Groupment object, I can get all the related GlobalContent objects without any problem.

**But if I switch the sides (inversedBy + JoinTable and MappedBy) the code only works the other way,** if I add/remove some GlobalContent from a Groupment object.

Any suggestion ? I've tried a shitload of things and nothing worked

**Comment by derrabus** [comment]
Yes, the relation is managed on the owning side and only on the owning side.

**Reply by toto33331** [reply]
So, even if I've followed the Doctrine doc to do a bidirectional ManyToMany it's only managable on one side.
And I can't properly manage relations from the other side.

I must do this a dirty way like compare pre and post form, and catch all the adds and removes to get each entity and add/remove from it ?

**Reply by derrabus** [reply]
> So, even if I've followed the Doctrine doc to do a bidirectional ManyToMany it's only managable on one side.

Yes.

> And I can't properly manage relations from the other side.

You would need to implement that on your own, e.g. by updating the owning side from the inverse side. Always keep in mind that the ORM only tracks the owning side only for change calculation.

> I must do this a dirty way like compare pre and post form, and catch all the adds and removes to get each entity and add/remove from it ?

That's one way to do it, but not the way I'd recommend.
